test/cpp_extensions/complex_registration_extension.cpp

```diff
@@ -46,10 +47,11 @@ static Tensor empty_complex(IntArrayRef size, const TensorOptions & options, c10
   return tensor;
 }
 
-static auto& complex_empty_registration = globalATenDispatch().registerOp(
-    TensorTypeId::ComplexCPUTensorId,
-    "aten::empty.memory_format(int[] size, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, MemoryFormat? memory_format=None) -> Tensor",
-    &empty_complex);
+static auto complex_empty_registration = torch::RegisterOperators()
+  .op(torch::RegisterOperators::options()
+    .schema("aten::empty.memory_format(int[] size, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, MemoryFormat? memory_format=None) -> Tensor")
+    .impl_unboxedOnlyKernel<decltype(empty_complex), &empty_complex>(TensorTypeId::ComplexCPUTensorId)
+    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA));
```

Comment:
The new API seems worse than the old one. As a pre-existing schema string that we defined, the custom registration shouldn't be in the business of specifying the alias analysis: the aliasing was predetermined by us, the library authors, user implementations of the contract don't get to change the contract.

I have also mentioned in 1:1 that I don't really want the code to be explicitly mentioning ComplexCPUTensorId, though this is a defect of the old code too.

Reply:
Both good points. Unrelated to this PR though. XLA and everything else is already using this new API, complex was somehow missed. We should fix the API but not block this PR by it.